modules/compose/compose_api.go

```diff
@@ -474,15 +465,14 @@ func (d *dockerCompose) lookupContainer(ctx context.Context, svcName string) (*t
 
 	ctr.SetProvider(dockerProvider)
 
+	d.containersLock.Lock()
```

Comment:
Totally right, makes it more sense to lock inside the lookup method. Good catch!